File: Arrays/3.TwoSum2/solution.js
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/

var twoSum = function(numbers, target) {
    let i = 0;
    let j = numbers.length-1;

  while(i < j){

    if(numbers[i] + numbers[j] == target){
        return [i+1,j+1];
    }
    else if(numbers[i] + numbers[j] > target){
        j--;
    }
    else{
        i++;
    }
  }  
};

let numbers = [2,7,11,15];
let target = 9;

console.log(twoSum(numbers,target));

File: Arrays/3.TwoSum2/solution.java
public class solution {
    
    public static int [] twosum(int[] numbers, int taraget ){
       
        int result[] = new int[2];
        int i = 0; 
        int j = numbers.length-1;
       
        while(i < j){

            if(numbers[i] + numbers[j] == taraget){
                result[0] = i+1;
                result[1] = j+1;
                return result;
            }

            if(numbers[i] + numbers[j] > taraget){
                j--;
            }
            else{
                i++;
            }
        }

        return result;
    }

    public static void main(String args[]){
        int numbers[] = {2,7,11,15};
        int  target = 9;
        int ans[] = twosum(numbers, target);
        System.out.println(ans[0]+" "+ans[1]);
    }
}
